hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/ContextualRetriever.java

```diff
@@ -41,6 +51,41 @@ private ContextualRetriever() {
         throw new UnsupportedOperationException("Utility class only");
     }
 
+    /**
+     * Returns the account if it exists and is usable. A {@link HandleException} is thrown if the account is invalid. Note that this method should also work with account ID's that represent smart contracts
+     *
+     * @param accountId the ID of the account to get
+     * @param accountStore the {@link ReadableTokenStore} to use for account retrieval
+     * @param autoRenewConfig the {@link AutoRenewConfig} to use for determining account expiry validation
+     * @throws HandleException if any of the account conditions are not met
+     */
+    public static Account getIfUsable(
+            @NonNull final AccountID accountId,
+            @NonNull final ReadableAccountStore accountStore,
+            @NonNull final AutoRenewConfig autoRenewConfig) {
+        requireNonNull(accountId);
+        requireNonNull(accountStore);
+        requireNonNull(autoRenewConfig);
+
+        final var acct = accountStore.getAccountById(accountId);
+        validateTrue(acct != null, INVALID_ACCOUNT_ID);
+        validateFalse(acct.deleted(), ACCOUNT_DELETED);
+        final var isSmartContract = acct.smartContract();
+        validateFalse(
+                acct.expiredAndPendingRemoval(),
+                isSmartContract ? CONTRACT_EXPIRED_AND_PENDING_REMOVAL : ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
+
+        final var expiryStatus = ExpiryValidator.getAccountOrContractExpiryStatus(
+                acct.tinybarBalance(),
+                false,
+                isSmartContract,
+                autoRenewConfig.expireContracts(),
+                autoRenewConfig.expireAccounts());
+        validateTrue(expiryStatus == OK, expiryStatus);
+
+        return acct;
```

Comment:
For all this you can use `expiryValidator.isDetached` or `expiryValidator.expirationStatus`

Reply:
Done